whetstone/compiler-test/src/test/kotlin/com/freeletics/mad/whetstone/codegen/TestHelper.kt

```diff
@@ -5,39 +5,118 @@ import com.freeletics.mad.whetstone.ComposeFragmentData
+private fun compile(fileName: String, source: String, output: String) {
+    val compilation = KotlinCompilation().apply {
+        configure()
+
+        sources = listOf(
+            sourceFile(fileName, source),
+            sourceFile(fileName.testFileName(), output),
+        )
+    }
+
+    assertThat(compilation.compile().exitCode).isEqualTo(ExitCode.OK)
+}
+
+private fun compileWithAnvil(fileName: String, source: String, output: String) {
+    val compilation = KotlinCompilation().apply {
+        configure()
+        configureAnvil()
+
+        sources = listOf(
+            sourceFile(fileName, source),
+        )
+    }
+
+    assertThat(compilation.compile().exitCode).isEqualTo(ExitCode.OK)
+    assertThat(compilation.generatedFile(fileName)).isEqualTo(output)
+}
+
+private fun KotlinCompilation.sourceFile(name: String, content: String): SourceFile {
+    val path = "${workingDir.absolutePath}/sources/src/main/kotlin/$name"
+    Files.createDirectories(File(path).parentFile!!.toPath())
+    return SourceFile.kotlin(path, content)
+}
+
+private fun KotlinCompilation.generatedFile(name: String): String {
+    val path = "${workingDir.absolutePath}/build/anvil/${name.testFileName()}"
+    return File(path).readText()
+}
+
+private fun String.testFileName(): String {
+    val path = substringBeforeLast("/")
+    val name = substringAfterLast("/")
+    return "$path/Whetstone$name"
+}
+
+private fun KotlinCompilation.configure() {
+    compilerPlugins = compilerPlugins + listOf(ComposeComponentRegistrar())
+    kotlincArguments = kotlincArguments + listOf(
+        "-P",
+        "plugin:androidx.compose.compiler.plugins.kotlin:suppressKotlinVersionCompatibilityCheck=1.7.22"
+    )
+    jvmTarget = "11"
+    inheritClassPath = true
+    messageOutputStream = System.out // see diagnostics in real time
+}
+
+private fun KotlinCompilation.configureAnvil() {
+    compilerPlugins = compilerPlugins + listOf(AnvilComponentRegistrar())
+
+    val anvilCommandLineProcessor = AnvilCommandLineProcessor()
+    commandLineProcessors = listOf(anvilCommandLineProcessor)
+
+    pluginOptions = listOf(
+        PluginOption(
+            pluginId = anvilCommandLineProcessor.pluginId,
+            optionName = "src-gen-dir",
+            optionValue = File(workingDir, "build/anvil").absolutePath
+        ),
+        PluginOption(
+            pluginId = anvilCommandLineProcessor.pluginId,
+            optionName = "generate-dagger-factories",
+            optionValue = "false"
+        ),
+        PluginOption(
+            pluginId = anvilCommandLineProcessor.pluginId,
+            optionName = "generate-dagger-factories-only",
+            optionValue = "false"
+        ),
+        PluginOption(
+            pluginId = anvilCommandLineProcessor.pluginId,
+            optionName = "disable-component-merging",
+            optionValue = "false"
         )
```

Comment:
Anvil has a helper for this in the utils test fixture but I got some weird compilation errors with that. I'll try again after the Kotlin 1.8/Anvil 2.4.4 update.